logstash-core/src/main/java/org/logstash/ackedqueue/Page.java

```diff
@@ -34,6 +34,8 @@ public Page(int pageNum, Queue queue, long minSeqNum, int elementCount, long fir
         this.lastCheckpoint = new Checkpoint(0, 0, 0, 0, 0);
         this.pageIO = pageIO;
         this.writable = writable;
+
+        assert this.pageIO != null : "invalid null pageIO";
```

Comment:
Do you think adding `@NotNull` annotation to the `pageIO` parameter could be a useful replacement for this assertion?

Reply:
AFAIK `@NotNull` will not result in run-time check and it does not enforce that the method cannot be called with a `null` argument. It is more for external tools. I can definitely add it.
Note that I am also on the fence about using an assertion to do parameter validation since assertions can be disabled. My thinking here is that I looked and could not see a case where a `null` `pageIO` would be passed but adding this assertion would make the test break if there is an oversight in future refactoring. It also pairs with the removal of the `null` checks in the `close()` and `purge()` methods. WDYT?

Reply:
added.